smart-contracts/assembly/contracts/FT/token-burn.ts

```diff
@@ -76,3 +76,33 @@ export function _decreaseTotalSupply(amount: u256): void {
 
   Storage.set(TOTAL_SUPPLY_KEY, u256ToBytes(newTotalSupply));
 }
+
+/**
+ * Burn tokens from the caller address
+ *
+ * @param binaryArgs - byte string with the following format:
+ * - the amount of tokens to burn obn the caller address (u256).
+ */
+export function burnFrom(binaryArgs: StaticArray<u8>): void {
+  const args = new Args(binaryArgs);
+  const amount = args
+    .nextU256()
+    .expect('amount argument is missing or invalid');
+  const account = new Address(
+    args.nextString().expect('account argument is missing or invalid'),
+  );
+
+  const spenderAllowance = _allowance(account, Context.caller());
+
+  assert(spenderAllowance >= amount, 'burnFrom failed: insufficient allowance');
+
+  _decreaseTotalSupply(amount);
+
+  _burn(account, amount);
+
+  _approve(account, Context.caller(), u256.sub(spenderAllowance, amount));
```

Comment:
Why aren't you using - operator ?
```suggestion
  _approve(account, Context.caller(), spenderAllowance - amount);
```

Reply:
Because the linter was not happy, but indeed I could have used the - operator.